src/hooks/spaces/useWorldSpaces.ts

```diff
@@ -0,0 +1,26 @@
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import { collection, query, where } from "firebase/firestore";
+
+import { ALWAYS_EMPTY_ARRAY, COLLECTION_SPACES } from "settings";
+
+import { AnyVenue } from "types/venues";
+
+import { withIdConverter } from "utils/converters";
+import { WithId } from "utils/id";
+
+export const useWorldSpaces = (worldId: string = ""): WithId<AnyVenue>[] => {
```

Comment:
Allows for extensibility and/or ignoring different fields, like: 
 - Add new filter field later on
 - Add some flags like preference to having initialData
 - Add default array if empty isn't always desired
 - ignore some old  fields, etc.

```suggestion
type UseWorldSpaces = (options:{ worldId: WorldId }) => { spaces: SpaceWithId };

export const useWorldSpaces: UseWorldSpaces = ({worldId = ""}) => {
```

Reply:
Was done in https://github.com/sparkletown/sparkle/pull/2820/commits/aeb889a3ddc83638a1abdc657bc571aec9bcb95d